rfd/00XX-ci.md

```diff
@@ -0,0 +1,83 @@
+
+Branch protection rules will be updated to [dismiss approvals](https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule#creating-a-branch-protection-rule) to `master` and all release branches if new changes are pushed to the PR after approval. To prevent a slowdown to developer productivity, `Require branches to be up to date before merging` will be disabled. Any breaking changes caused by this will be caught by automated testing that will run against `master` and all release branches upon merge.
+
+#### Workflow Permissions
+
+Workflows will run either in the context of `master` or `HEAD` of the PR and [with access to secrets or without](https://docs.github.com/en/actions/reference/encrypted-secrets#accessing-your-secrets). Use cases are illustrated in the `Use Cases and Permissions` section.
+
+### Use Cases and Permissions
+
+#### PR Approval Bot
+
+Context: `master`, Secrets: `GITHUB_TOKEN: READ`
+
+A bot that approves PRs should not run in the context of the PR, otherwise an attacker could simply change the approval workflow itself, and approve their own PR. A PR approval bot should run in the context of `master`.
+
+In addition, because an approval bot does not need to make changes to the repository, it should not have access to any additional secrets and should only have `GITHUB_TOKEN: READ` permissions.
+
+#### PR Assignment Bot
+
+Context `master`, Secrets: `GITHUB_TOKEN: WRITE`
+
+A bot that approves PRs should not run in the context of the PR, otherwise an attacker could simply change the approval workflow itself, and approve their own PR. A PR approval bot should run in the context of `master`.
+
+Because a PR assignment bot needs to update the repository (to assign reviewers), it must have `GITHUB_TOKEN: WRITE` access, however, this is safe because the only user controlled input would be the PR assigners GitHub login.
+
+#### Posting Status to Slack
+
+Context `master`, Secrets: `SLACK: WRITE`
+
+A bot that posts an update to Slack should not in the context of a PR, otherwise an attacker could potentially gain access to the secret. A bot that needs a Slack token should run in the context of `master`. It should read from an artifact uploaded by a unprivileged workflow using [actions/upload-artifact](https://github.com/actions/upload-artifact)and post that to Slack.
+
+#### Running Tests
+
+Context `HEAD` of PR, Secrets: `GITHUB_TOKEN: READ`
+
+Running tests must occur in the context of `HEAD` of the PR to test new changes. Due to this, this workflow should only have access to `GITHUB_TOKEN: READ` and no access to any additional secrets.
```

Comment:
There may be a secret:

https://github.com/gravitational/teleport/blob/d12a4a6c562b3978619998d9843e0ce01ff998c5/.drone.yml#L175-L176

If you're going to replace this, it'd be worth discussing how. "We're not running that test on PRs" is a ok answer by me -- but then it loops back to the discussion of "will post-merge validation be stricter than pre-merge validation?"